### PR TITLE
added help page

### PR DESCRIPTION
**Pull request by madhavmansuriya40**
Screenshot 2020-09-30 at 5 41 20 PM
Screenshot 2020-09-30 at 5 41 09 PM
Screenshot 2020-09-30 at 5 49 16 PM
Screenshot 2020-09-30 at 5 49 04 PM
Screenshot 2020-09-30 at 5 41 47 PM
Screenshot 2020-09-30 at 5 41 29 PM

### CODE REVIEW

File: src/app/core/services/extended-org-users.service.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { ApiService } from './api.service';
+import { map, switchMap, concatMap, reduce } from 'rxjs/operators';
+import { ExtendedOrgUser } from '../models/extended-org-user.model';
+import { DataTransformService } from './data-transform.service';
+import { range, Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExtendedOrgUsersService {
```
Comment (Julias0) [inline src/app/core/services/extended-org-users.service.ts:11]: These methods already exist inside `OrgUserService`
Remove this file and use those

File: src/app/core/services/org-user.service.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export class OrgUserService {
 
   constructor(
     private apiService: ApiService,
-    private authService: AuthService
+    private authService: AuthService,
```
Comment (Julias0) [inline src/app/core/services/org-user.service.ts:14]: why @madhavmansuriya40 why

File: src/app/fyle/help/help.page.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<ion-header>
+  <ion-toolbar class="header-container">
```
Comment (Julias0) [inline src/app/fyle/help/help.page.html:2]: help-content--header-container

File: src/app/fyle/help/help.page.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+</ion-header>
+
+<ion-content class="help-content">
+  <div class="help-content--helper-buttons">
+    <div>
+      <button (click)="openContactSupportDialog()" mat-flat-button
+        class="help-content--helper-buttons--btn-help">Contact Support</button>
+    </div>
+    <div>
+      <button (click)="openHelpLink()" mat-flat-button class="help-content--helper-buttons--btn-help">Explore Knowledge
```
Comment (Julias0) [inline src/app/fyle/help/help.page.html:14]: this is not proper BEM

File: src/app/fyle/help/help.page.scss [hunk 5]
```diff
@@ -0,0 +1,84 @@
+$brand-secondary-primary: #220033;
```
Comment (Julias0) [inline src/app/fyle/help/help.page.scss:1]: spacing should be 2 spaces - and use proper BEM

You cant have nested elements in BEM

refer to the other pages

File: src/app/fyle/help/help.page.scss [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+				border-bottom: 1px solid #ccc;
+				margin-bottom: 10px;
+			}
+
+			&--sub-title {
+				font-size: 14px;
+				height: 20px;
+				overflow: hidden;
+				white-space: nowrap;
+				text-overflow: ellipsis;
```
Comment (Julias0) [inline src/app/fyle/help/help.page.scss:80]: use ellipses pipe instead of this

File: src/app/fyle/help/help.page.scss [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+				text-align: center;
+				justify-content: center;
+				line-height: 1.2;
+				border-bottom: 1px solid #ccc;
+				margin-bottom: 10px;
+			}
+
+			&--sub-title {
+				font-size: 14px;
+				height: 20px;
```
Comment (Julias0) [inline src/app/fyle/help/help.page.scss:77]: use only max and min height/ width

File: src/app/fyle/help/help.page.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      );
+      this.loadingController.dismiss();
+      // only need to display 5 admins, so spliced 5
+      this.presentSupportModal('contact_support', orgAdmins.splice(0, 5));
+    });
+  }
+
+  async presentLoading() {
+    // TODO: to replace with new loader, suggested to create component
+    const loading = await this.loadingController.create({
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:39]: use loaderService

File: src/app/fyle/help/help.page.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+export class HelpPage implements OnInit {
+
+  constructor(
+    private modalController: ModalController,
+    private extendedOrgUserService: ExtendedOrgUsersService,
+    private loadingController: LoadingController
+    ) { }
+
+  openContactSupportDialog() {
+    this.presentLoading();
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:24]: this approach is incorrect - use loader service - refer to how other pages have used it

File: src/app/fyle/help/help.page.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    // getting admins list, required for contact support modal
+    let orgAdmins = [];
+    this.extendedOrgUserService.getAllCompanyEouc().subscribe(user => {
+      orgAdmins = user.filter(user =>
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:28]: do this inside the pipe with map

File: src/app/fyle/help/help.page.ts [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        user.ou.roles.indexOf('ADMIN') > -1 && user.ou.status === 'ACTIVE'
+      );
+      this.loadingController.dismiss();
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:31]: loaderService

File: src/app/fyle/help/help.page.ts [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+  async presentLoading() {
+    // TODO: to replace with new loader, suggested to create component
+    const loading = await this.loadingController.create({
+    });
+    await loading.present();
+  }
+
+  openLogMileageDialog() {
+    this.presentSupportModal('log_mileage', '');
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:45]: why is the second argument an empty string?

File: src/app/fyle/help/help.page.ts [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  openCaptureEmailReceiptsDialog() {
+    this.presentSupportModal('capture_email', '');
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:49]: why is the second argument an empty string?

File: src/app/fyle/help/help.page.ts [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  openManageGoIbiboBookingsDialog() {
+    this.presentSupportModal('goibibo_support', '');
```
Comment (Julias0) [inline src/app/fyle/help/help.page.ts:53]: why is the second argument an empty string?

File: src/app/fyle/help/support-dialog/support-dialog.page.html [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+    <ion-buttons slot="end" style="display: block">
+      <ion-button color="dark" (click)="closeDialog()">
+        <ion-icon name="close-outline"></ion-icon>
+      </ion-button>
+    </ion-buttons>
+  </ion-toolbar>
+</ion-header>
+
+<ion-content class="contact-support">
+  <!-- contact support dialog -->
```
Comment (Julias0) [inline src/app/fyle/help/support-dialog/support-dialog.page.html:12]: remove all comments

File: src/app/fyle/help/support-dialog/support-dialog.page.html [hunk 16]
```diff
@@ -0,0 +1,96 @@
+<ion-header>
+  <ion-toolbar>
+    <ion-buttons slot="end" style="display: block">
```
Comment (Julias0) [inline src/app/fyle/help/support-dialog/support-dialog.page.html:3]: no inline styling

File: src/app/fyle/help/support-dialog/support-dialog.page.html [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+    <div class="contact-support--header">
+      <div class="contact-support--header--title">Contact Support</div>
+    </div>
+    
+    <!-- contact support content -->
+    <div class="contact-support--content">
+      <p class="contact-support--content--heading">Having trouble with the process?</p>
+      <p>contact your admin</p>
+      
+      <div class="contact-support--content--admin-list">
```
Comment (Julias0) [inline src/app/fyle/help/support-dialog/support-dialog.page.html:24]: use proper BEM

File: src/app/fyle/help/support-dialog/support-dialog.page.scss [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+
+
+ &--content {
+    overflow: scroll;
+    margin: 30px;
+
+    &--helpLinkBtn {
+      width: 100%;
+			font-weight: 400;
+			height: 50px;
```
Comment (Julias0) [inline src/app/fyle/help/support-dialog/support-dialog.page.scss:27]: arrey indentation :(

File: src/app/fyle/help/support-dialog/support-dialog.page.scss [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+    &--content-container {
+      padding: 20px 0;
+      font-size: 16px;
+      line-height: 24px;
+    }
+
+    &--note-container {
+      border-radius: 4px;
+      padding: 10px;
+      background-color: #f1f1f1;
```
Comment (Julias0) [inline src/app/fyle/help/support-dialog/support-dialog.page.scss:63]: colors in variables everywhere please

**Review by Julias0 — CHANGES_REQUESTED** [review]
Fix and ping back

File: src/app/fyle/help/help.page.scss (hunk 5 above)
Comment (madhavmansuriya40) [reply src/app/fyle/help/help.page.scss:1]: indentations are correct in code, problem here (can ignore)

**Review by Julias0 — APPROVED** [review]

**Pull request closed without merging**